src/modm/driver/radio/dw3110/dw3110_phy_impl.hpp

```diff
@@ -0,0 +1,1415 @@
+/*
+ * Copyright (c) 2024, Elias H.
+ * Copyright (c) 2024, Raphael Lehmann
+ * Copyright (c) 2024, Michael Jossen
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#ifndef MODM_DW3110_PHY_HPP
+#error "Don't include this file directly, use 'dw3110_phy.hpp' instead!"
+#endif
+
+#include <modm/debug/logger.hpp>
+
+#include "dw3110_definitions.hpp"
+#include "dw3110_phy.hpp"
+
+template<typename SpiMaster, typename Cs>
+modm::Dw3110Phy<SpiMaster, Cs>::Dw3110Phy()
+{
+	this->attachConfigurationHandler([]() {
+		SpiMaster::setDataMode(SpiMaster::DataMode::Mode0);
+		SpiMaster::setDataOrder(SpiMaster::DataOrder::MsbFirst);
+	});
+	Cs::setOutput(true);
+}
+
+template<typename SpiMaster, typename Cs>
+modm::ResumableResult<bool>
+modm::Dw3110Phy<SpiMaster, Cs>::initialize(Dw3110::Channel channel, Dw3110::PreambleCode pcode,
+										   Dw3110::PreambleLength plen,
+										   Dw3110::StartFrameDelimiter sfd)
+{
+	using namespace std::chrono_literals;
+	RF_BEGIN();
+
+	RF_CALL(fetchChipState());
+	if (chip_state == Dw3110::SystemState::TX || chip_state == Dw3110::SystemState::RX ||
+		chip_state == Dw3110::SystemState::TX_WAIT || chip_state == Dw3110::SystemState::RX_WAIT)
+	{
+		RF_CALL(sendCommand<Dw3110::FastCommand::CMD_TXRXOFF>());
+	}
+
+	// Loop until valid state
+	timeout.restart(1ms);
+	while (chip_state != Dw3110::SystemState::IDLE_RC &&
+		   chip_state != Dw3110::SystemState::IDLE_PLL)
+	{
+		if (timeout.execute())
+		{
+			MODM_LOG_ERROR << "Timeout waiting for IDLE State!" << (int)chip_state << modm::endl;
```

Comment:
`std::to_underlying(chip_state)` is less error-prone than the C-style cast. It's impossible to get the type wrong.

Reply:
to_underlying would resolve to uint8_t here which would be printed as char not as int, so casting to int was desired.

Reply:
The modm streams only print `char` as a character but `unsigned char` (`uint8_t`) and `signed char` (`int8_t`) as integers. 

```c++
MODM_LOG_ERROR << uint8_t(100) << ", " << int8_t(100) << ", " << char(100);
```
prints `100, 100, d`.

You can leave the code as is. It's not critical here.